src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+
+  const separator = fromFormatCopy.pop();
+  const parts = date.split(separator);
+
+  const dateParts = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    dateParts[part] = parts[index];
+  });
+
+  if (toFormatCopy.includes('YY') && dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  if (toFormatCopy.includes('YYYY') && dateParts['YY']) {
+    dateParts['YYYY'] =
+      dateParts['YY'] < '30' ? '20' + dateParts['YY'] : '19' + dateParts['YY'];
+  }
+
+  const newSeparator = toFormatCopy.pop();
+
+  const newDate = toFormatCopy
+    .map((part) => dateParts[part])
+    .join(newSeparator);
```

Comment:
The use of `newSeparator` as a delimiter for joining date parts may not align with the expected format in `toFormat`. Ensure that the delimiter matches the intended output format.